sport/app/rugby/jobs/RugbyStatsJob.scala

```diff
@@ -1,57 +1,60 @@
 package rugby.jobs
 
-import common.{Jobs, AkkaAgent, ExecutionContexts, Logging}
+import common.{AkkaAgent, ExecutionContexts, Logging}
 import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
 import rugby.feed.{OptaFeed, RugbyOptaFeedException}
 import rugby.model.Match
+import scala.concurrent.Future
 
-object RugbyStatsJob extends ExecutionContexts with Logging {
-  private val liveScoreMatches = AkkaAgent[Seq[Match]](Seq.empty)
-  private val fixturesAndResultsMatches = AkkaAgent[Seq[Match]](Seq.empty)
-  val dateFormat: DateTimeFormatter = DateTimeFormat.forPattern("yyyy/MM/dd")
+object RugbyStatsJob extends RugbyStatsJob
 
+trait RugbyStatsJob extends ExecutionContexts with Logging {
+  protected val liveScoreMatches = AkkaAgent[Map[String, Match]](Map.empty)
+  protected val fixturesAndResultsMatches = AkkaAgent[Map[String, Match]](Map.empty)
+  val dateFormat: DateTimeFormatter = DateTimeFormat.forPattern("yyyy/MM/dd")
 
   def run() {
-
-    liveScores
-    fixturesAndResults
+    sendLiveScores(OptaFeed.getLiveScores)
+    fixturesAndResults(OptaFeed.getFixturesAndResults)
   }
 
-  def liveScores {
-    OptaFeed.getLiveScores.map { matches =>
-      matches.map { aMatch =>
-        val date = dateFormat.print(aMatch.date)
-        val key = s"$date/${aMatch.homeTeam.id}/${aMatch.awayTeam.id}"
-        liveScoreMatches.send {_ :+ aMatch}
-      }
-
+  def sendLiveScores(scoreData: Future[Seq[Match]]) : Future[Any] = {
```

Comment:
I wanted it for testing `sendLiveScores`, not that I cared what the return value of the agent send was (which will be set to the value of whatever you passed in)